### DISCUSSION TITLE
Inherit project theme when overriding theme properties

### DISCUSSION THREAD

**Discussion by Benguin** [godotengine/godot-proposals#4513] [GUI]
Currently (4.x), when creating style properties as theme overrides on a control node, new properties are created with fully default values, even if the project's default theme has been edited.

It could save a lot of time if instead of creating a 'blank' theme property, it would copy the theme property that the control would otherwise inherit from the theme / default theme. That way, minor adjustments can be made to the defaults which have been set, rather than having to completely restyle the Control from scratch, or go diving into the theme to copy the correct style.

**Example scenario:**
The developer has created a theme which sets the colours, margins, borders, etc on the Panel element. Some panels in the layout built with the theme need all corners to be rounded, and others need a subset of corners to be rounded depending on where it is used, but they otherwise have identical styling.

The developer would add a panel to the scene, override the StyleBoxFlat for its 'Panel' style, then modify the border radii for that control; the control's background color, border color, border width and etc would all already match the values from the current inherited theme, saving time and preventing any inconsistencies from being introduced in the theme.

**Comment by Calinou** [comment]
This is a good idea, but I'm not sure about how this could be implemented (since the default StyleBoxFlat properties would be changed only within the editor, and depending on the base StyleBoxFlat already used in the theme).

**Comment by theDoktorJ** [comment]
I am running into this limitation as well.  I need to change to rounded corners for a scene that inherits from Panel, and even with overrides, I have to override the entire panel theme, when all I want are rounded corners.  Would it be possible to could split the panel theme into multiple styleboxes, one for colors, one for rounded edges, and so on, and just have the options that are not set and/or not used, be overridden by the project or inherited theme?

**Reply by Calinou** [reply]
Using [resource state inheritance](https://github.com/godotengine/godot/pull/86779) sounds like a better solution to me, although it would need to be expanded to allow inheriting from default theme resources.
